src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php

```diff
@@ -888,11 +888,15 @@ protected function typeMultiPolygonZ(Fluent $column)
      */
     private function formatPostGisType(string $type, Fluent $column)
     {
-        if ($column->isGeometry !== null) {
-            return "geometry($type".($column->projection === null ? '' : ", $column->projection").')';
+        if ($column->isGeometry === null) {
+            return sprintf('geography(%s, %s)', $type, $column->projection ?? '4326');
         }
 
-        return "geography($type, ".($column->projection ?? '4326').')';
+        if ($column->projection !== null) {
+            return sprintf('geometry(%s, %s)', $type, $column->projection);
+        }
+
+        return "geometry({$type})";
```

Comment:
Do you need this? Above you are checking if it's null, and then check if it's not null.

Reply:
Different property.